lib/farscape/agent.rb

```diff
@@ -23,10 +39,19 @@ def representor
       safe? ? SafeRepresentorAgent : RepresentorAgent
     end
 
-    def enter(entry = entry_point)
-      @entry_point ||= entry
+    # Discovers provided a key and template_variables.
+    # This method is here to be easily overwritten or monkey-patched if needed.
+    def discover_entry_point(key, template_variables = {})
+      Discovery.new.discover(self.class.config, key, template_variables)
+    end
+
+    def enter(entry = entry_point, template_variables = {})
       raise "No Entry Point Provided!" unless entry
-      response = client.invoke(url: entry, headers: get_accept_header(media_type))
+      @entry_point ||= entry
+      unless URI::DEFAULT_PARSER.regexp[:ABS_URI] =~ @entry_point
```

Comment:
since we're depending on addressable I think this should use it here instead of those cryptic constants in the stdlib's URI, for instance:
```ruby
Addressable::URI.parse("http://www.google.com").absolute?
# => true
Addressable::URI.parse("http://www.google.com").relative?
# => false
```